client/signup/navigation-link/test/index.jsx

```diff
@@ -45,42 +50,43 @@ describe( 'NavigationLink', () => {
 	} );
 
 	test( 'should render Button element', () => {
-		const wrapper = shallow( <NavigationLink { ...props } /> );
+		const { container } = render( <NavigationLink { ...props } /> );
 
-		expect( wrapper.find( 'ForwardRef(Button)' ) ).toHaveLength( 1 );
-		expect( wrapper.find( 'ForwardRef(Button)' ).props().className ).toEqual( 'navigation-link' );
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 1 );
+		expect( container.firstChild.nodeName ).toBe( 'BUTTON' );
+		expect( container.firstChild ).toHaveClass( 'navigation-link' );
 	} );
 
 	test( 'should render no icons when the direction prop is undefined', () => {
-		const wrapper = shallow( <NavigationLink { ...props } /> );
-		expect( wrapper.find( Gridicon ).exists() ).toEqual( false );
+		render( <NavigationLink { ...props } /> );
+		expect( screen.queryByTestId( /gridicon-/ ) ).not.toBeInTheDocument();
 	} );
 
 	test( 'should render right-arrow icon when the direction prop is "forward".', () => {
-		const wrapper = shallow( <NavigationLink { ...props } direction="forward" /> );
+		const { container } = render( <NavigationLink { ...props } direction="forward" /> );
 
-		expect( wrapper.find( Gridicon ) ).toHaveLength( 1 );
-		expect( wrapper.childAt( 1 ).props().icon ).toEqual( 'arrow-right' );
-		expect( wrapper.childAt( 0 ).text() ).toEqual( 'translated:Skip for now' );
+		expect( container.firstChild ).toHaveTextContent( 'translated:Skip for now' );
```

Comment:
We could instead:

```suggestion
		expect( screen.queryByText( 'translated:Skip for now' ) ).toBeVisible();
```

As the library suggests we should use `screen` to query elements whenever possible.